executor/aggfuncs/func_group_concat.go

```diff
@@ -103,8 +103,11 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 			p.buffer.WriteString(e.sep)
 		}
 	}
-	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
```

Comment:
`group_concat distinct` doesn't panic but got the wrong result.

```
mysql> select group_concat(distinct a) from t;
+--------------------------+
| group_concat(distinct a) |
+--------------------------+
|                          |
+--------------------------+
1 row in set (0.00 sec)
```
in mysql return null

Reply:
I can reproduce this via queries provided in this PR. Maybe you need double-check?